### ISSUE TITLE
Issue with Adult Trading Items disappearing

### ISSUE THREAD

**Issue by iadlast** [gamestabled/OoT3D_Randomizer#332]
While I was playing with the Adult Link trading sequence items shuffled into the item pool, I noticed that I had the Prescription at one point, but after I picked up the Claim Check and gave it to Biggoron, the Prescription disappeared from my L<->R trading item rotation, and it didn't fix itself even after I picked up another trading item (the Odd Mushroom).

Maybe the delivery of the Claim Check causes all other currently owned trading items to disappear due to the game thinking the quest has been completed?

**Comment by cjohnson57** [comment]
Thanks for the report! That would make sense if that was the case.

**Comment by gamestabled** [comment]
I'll have to see if I can recreate this, thanks for posting it.

Maybe relatedly, maybe not, but just so we have it documented, this is a quote from freddy in the discord:

> So, apparently if you have Eyeball Frog with shuffled adult trade sequence, and then you grab the MS, you lose the frog, even if it wasn't the "active" item in the slot at that moment.
The only possible reason I can imagine for this is that the unk flags for the DMT scene aren't actually unused, but I could be wrong

To explain the last piece of that, the "unk" scene flags for DMT are used by the shuffled adult trade setting to keep track of which items are currently owned.

**Comment by gamestabled** [comment]
@iadlast I think I've figured out a likely cause of this issue. By chance, do you remember if you got the "Sheik in Kakariko" (vanilla nocturne) item location in between picking up the Prescription and trading the Claim Check?

**Comment by iadlast** [comment]
Hmm, I'm not sure, but I don't think I did. The generated seed made me run around a lot before I could gather the required medallions to trigger that item location.

If it helps, the hash for the seed in question is this one:
-Heart Container
-Hover Boots
-Beans
-Fairy Ocarina
-Map

**Comment by gamestabled** [comment]
this [commit](https://github.com/gamestabled/OoT3D_Randomizer/pull/333/commits/430d427bb0bb72564fcf448098bc080e9d7f0acd) is the fix for what i mentioned above; i think this is almost certainly a fix for what freddy experienced. It might also be the fix for what iadlast experienced, which I'm not sure how to recreate without the situation i mentioned above with the Sheik at Kakariko location. If enough time and playthroughs go by with no similar issue, it's probably fixed.

**Comment by iadlast** [comment]
Thank you very much for the prompt fix!

I'll try to replicate the conditions in my next run to see if the issue was solved, and I'll report my findings as soon as possible.

**Comment by iadlast** [comment]
Umm, I tried a new run with the latest nightly in which I ended up starting the game as Adult Link and in which I got the Odd Poultice from a large chest in Mido's House. However, I can't deliver it to Fado because Grogg is still in the usual spot in the Lost Woods instead of her.

The hash for this seed is:
-Frog
-SOLD OUT
-Bottled Fish
-Silver Gauntlets
-Longshot

I can also export the save file with the issue and upload it here if needed.

**Comment by iadlast** [comment]
I have found a related issue: having the Eye Drops in the inventory makes it impossible for the Professor to give the reward for the Laboratory PoH item location, as he just repeats his spiel about wanting a souvenir, and I had to restore a backup save to be able to go beyond where I was, as I was stuck as Adult Link with no way to access any areas and no way to deliver the Eye Drops due to the way to Biggoron being blocked by boulders and not having access to Bombchus, the Megaton Hammer or the Bomb Bag.

EDIT: I've also found another issue, though this isn't one that I can consistently reproduce: sometimes when speaking to Mutoh after giving him the Poacher's Saw, the game softlocks on his animations after his textboxes end but before he gives the reward for the Saw. I also need to mention I have on my inventory all the Adult trading items from the Pocket Cucco to the Saw. Hash for this one is Big Magic, Shard of Agony, SOLD OUT, Frog and Kokiri Tunic; and I have backed up the save for testing it.

**Comment by iadlast** [comment]
Okay, leaving aside the two issues I posted some hours ago in the previous comment, the one about the items vanishing has been solved! I tried both giving Biggoron the Claim Check with several other items in trading items in tow (Pocket Cucco, Cojiro, Odd Mushroom and Odd Poultice) before triggering the Sheik in Kakariko item location, and after triggering it. In both cases, the four items before the Claim Check are preserved properly! Good job!

**Comment by gamestabled** [comment]
> 
> EDIT: I've also found another issue, though this isn't one that I can consistently reproduce: sometimes when speaking to Mutoh after giving him the Poacher's Saw, the game softlocks on his animations after his textboxes end but before he gives the reward for the Saw. I also need to mention I have on my inventory all the Adult trading items from the Pocket Cucco to the Saw. Hash for this one is Big Magic, Shard of Agony, SOLD OUT, Frog and Kokiri Tunic; and I have backed up the save for testing it.

(EDIT) HylianFreddy actually identified whats going on here, and theres a [video](https://youtu.be/Byx5vX4k2Lg) about it in oot64 (its a base game bug). But, quick text is going to make this much more likely to happen, so we should look into a solution nonetheless

The Lab Scientist issue should be fixed by [this commit](https://github.com/gamestabled/OoT3D_Randomizer/commit/3a39d576e1b4cdf4d2e9b52f95d5c91c8ca1f3e7)

**Comment by gamestabled** [comment]
The bug with the carpenter boss is fixed now by [this commit](https://github.com/gamestabled/OoT3D_Randomizer/commit/5d98ee4907ba5ddaa82448078742e3ed37a2e4bf). Seems like everything in this thread is resolved now so I'll close it
